bundles/net.tourbook/src/net/tourbook/weather/openweathermap/OpenWeatherMapRetriever.java

```diff
@@ -226,6 +240,24 @@ private String buildWeatherApiRequest(final long date) {
       return weatherRequestWithParameters.toString();
    }
 
+   private boolean canMakeRequest() {
+
+      final String lastRequestDayString = _state.get(STATE_LAST_REQUEST_DAY);
+      final LocalDate lastRequestDay = lastRequestDayString != null ? LocalDate.parse(lastRequestDayString) : LocalDate.of(2000, 1, 1);
+      int requestCount = getRequestCount();
```

Comment:
You should use the `net.tourbook.common.util.Util.getState...()` methods, this is also available for date values

Reply:
>  this is also available for date values

It is but t[he function](https://github.com/mytourbook/mytourbook/blob/878eef4605f5bc6632430dd8431cae2e3811376d/bundles/net.tourbook.common/src/net/tourbook/common/util/Util.java#L1154) needs a `dateTimeControl`which I don't have in this case. Instead, I used the `getStateString` function

Reply:
Why not extend these tools, e.g.

```
public static LocalDate getStateDate(final IDialogSettings state,
                                        final String stateKey,
                                        final LocalDate defaultValue) {

      final String value = state.get(stateKey);

      try {

         return LocalDate.parse(value);

      } catch (final Exception e) {

         return defaultValue;
      }
   }

   public static LocalDate getStateDate(final IDialogSettings state,
                                        final String stateKey,
                                        final LocalDate defaultValue,
                                        final DateTime dateTimeControl) {

      final LocalDate parsedValue = getStateDate(state, stateKey, defaultValue);

      dateTimeControl.setYear(parsedValue.getYear());
      dateTimeControl.setMonth(parsedValue.getMonthValue() - 1);
      dateTimeControl.setDay(parsedValue.getDayOfMonth());

      return parsedValue;
   }
```

Reply:
>Why not extend these tools, e.g.

Usually, I extend tools whenever a function is used at several places and not only once.

But since you're asking, I have added the function per your suggestion